src/api.cu

```diff
@@ -167,128 +170,163 @@ TRITONBACKEND_ModelInstanceExecute(
 {
   uint64_t all_start_time =
       std::chrono::steady_clock::now().time_since_epoch().count();
-  std::size_t total_inference_count = 0;
-  std::vector<TRITONBACKEND_Response*> responses;
+
+  std::vector<TRITONBACKEND_Request*> requests(
+      raw_requests, raw_requests + request_count);
+
+  auto instance_state = get_instance_state<ModelInstanceState>(*instance);
+  auto model_state = instance_state->StateForModel();
+  auto target_memory = get_native_memory_for_instance(instance_state->Kind());
+
+  // Get all input data and bail out gracefully if this fails
+  std::vector<InputBatch<float>> input_batches;
   try {
-    auto instance_state = get_instance_state<ModelInstanceState>(*instance);
-    ModelState* model_state = instance_state->StateForModel();
-    auto target_memory = get_native_memory_for_instance(instance_state->Kind());
+    input_batches = get_input_batches<float>(
+        static_cast<uint32_t>(0), requests, target_memory,
+        instance_state->get_raft_handle());
+  }
+  catch (TritonException& setup_err) {
+    return setup_err.error();
```

Comment:
In which case do we have to use `release_requests`? I see it used after, but I don't see why it would not result in a double free as well ?

Reply:
Since we are returning a non-`nullptr` error, the server releases requests itself, as shown [here](https://github.com/triton-inference-server/server/blob/main/src/backends/backend/triton_model.cc#L219). That being said, this seems to contradict the docs [here](https://github.com/triton-inference-server/backend#tritonbackend_request), so I'll open a docs issue against Triton.

Reply:
Opened docs issue here: https://github.com/triton-inference-server/server/issues/3176